logstash-core/src/main/java/org/logstash/config/ir/compiler/DatasetCompiler.java

```diff
@@ -298,6 +303,22 @@ private static MethodLevelSyntaxElement callFilterFlush(final ClassFields fields
         );
     }
 
+    private static MethodLevelSyntaxElement unsetPluginIdForLog4j() {
+        return () -> "org.apache.logging.log4j.ThreadContext.remove(\"plugin.id\")";
+    }
+
+    private static MethodLevelSyntaxElement setPluginIdForLog4j(final AbstractFilterDelegatorExt plugin) {
+        final ThreadContext context = RubyUtil.RUBY.getCurrentContext();
+        final IRubyObject configName = plugin.getConfigName(context);
+        return () -> "org.apache.logging.log4j.ThreadContext.put(\"plugin.id\", \"" + configName + "\")";
+    }
+
+    private static MethodLevelSyntaxElement setPluginIdForLog4j(final AbstractOutputDelegatorExt plugin) {
```

Comment:
this code can probably be DRY'ed up for both method signatures?

Reply:
You are right, I try lower the repetition, the two methods are identical but one work on `AbstractFilterDelegatorExt` and the other on `AbstractOutputDelegatorExt` and the two classes hasn't got any common ancestor